src/SFML/Graphics/Font.cpp

```diff
@@ -64,15 +58,12 @@ sfFont* sfFont_createFromStream(sfInputStream* stream)
 {
     CSFML_CHECK_RETURN(stream, nullptr);
 
-    sfFont* font = new sfFont;
-    font->Stream = CallbackStream(stream);
-    if (!font->This.loadFromStream(font->Stream))
-    {
-        delete font;
-        font = nullptr;
-    }
+    CallbackStream sfmlStream(stream);
+    auto font = sf::Font::loadFromStream(sfmlStream);
+    if (!font)
+        return nullptr;
 
-    return font;
+    return new sfFont{std::move(*font), {}, sfmlStream};
```

Comment:
This is not correct: `sfmlStream` object passed by reference to `loadFromStream` must remain valid while the `sf::Font` instance is alive, but it is going out of scope.
Not sure what is best to solve this. An idea is having an `std::unique_ptr<CallbackStream>` in `sfFont`.

Reply:
Looking at this function before and after, I'm not sure how anything about ownership changes. If you're correct then that would imply the state of this function on `master` also exhibits this issue.

Reply:
No, this is different. On master, the stream passed to `loadFromFile` which is `font->Stream` remains valid. It is owned by the `sfFont` instance.

Reply:
I'm confused. `sfmlStream` is used to construct the `sfFont` instance so the `sfFont` still owns it, does it not?

Reply:
`sfFont` is constructed with a copy of `sfmlStream`. The original stack variable `sfmlStream`, to which `sf::Font` internally has a pointer to and expects to remain valid, will become invalid at the end of the function.

Reply:
I think I'm following now. Good catch. Off the top of my head I can't picture a good way to fix this. Do you have any ideas?

Reply:
This discussion makes me realize `sfFont_copy` was probably wrong: making a copy of the `CallbackStream` instance while the `sf::Font` copy will still internally point to the original instance.

Anyways, I think this would be a correct solution: https://github.com/SFML/CSFML/compare/sfml3_updates...kimci86:CSFML:sfml3_updates. Not sure if there is some smarter solution.

Reply:
This makes sense to me. I think it's a correct solution.